StyleCop.Analyzers/StyleCop.Analyzers/Helpers/IndentationHelper.cs

```diff
@@ -92,19 +91,23 @@ public static SyntaxTrivia GenerateWhitespaceTrivia(IndentationOptions indentati
 
         private static int GetIndentationSteps(IndentationOptions indentationOptions, SyntaxTriviaList leadingTrivia)
         {
-            var indentationString = string.Empty;
-            for (var i = leadingTrivia.Count - 1; (i >= 0) && leadingTrivia[i].IsKind(SyntaxKind.WhitespaceTrivia); i--)
+            SyntaxTriviaList.Reversed reversed = leadingTrivia.Reverse();
+            int indentationCount = 0;
+
+            foreach (SyntaxTrivia trivia in reversed)
             {
-                indentationString = string.Concat(leadingTrivia[i].ToFullString(), indentationString);
+                if (!trivia.IsKind(SyntaxKind.WhitespaceTrivia))
+                {
+                    break;
+                }
+
+                foreach (char c in trivia.ToFullString())
+                {
+                    indentationCount += c == '\t' ? indentationOptions.TabSize : 1;
+                }
             }
 
-            var indentationCount = indentationString.ToCharArray().Sum(c => IndentationAmount(c, indentationOptions));
             return indentationCount / indentationOptions.IndentationSize;
```

Comment:
Suppose you have a line indented by 3 spaces, but the indentation size is 4. The current implementation will return 0 (i.e. truncation), but my implementation would return 1 (i.e. a ceiling function).